### DISCUSSION TITLE
command to install additional Applications??

### DISCUSSION THREAD

**Discussion by grand-lotus-iroh** [cockpit-project/cockpit#16566] [Q&A] [answered]
Cockpit version: 215
OS: Ubuntu 20.04 x64

When I goto the /apps page I only see Storage but I would very much like to check out Software Updates (PackageKit) and Network. May you please advise me on command I need to type?

How do I install application? I can't for the love of god find any documentation. I have a brand new fresh installation of Ubuntu on a VPS and I connected VIA PuTTY and typed in `sudo apt-get install cockpit` and access the browser no problem.

* https://cockpit-project.org/guide/latest/ (nothing when I CTRL + F)

**Comment by KKoukiou** [comment, marked as answer]
You can check the package names of the plugins available also https://cockpit-project.org/applications.html

Then install them with apt.  

However the applications page should also show them. Try hitting the 'refresh' icon button on the upper right corner, I recall there was a bug on that page.

**Comment by garrett** [comment]
@grand-lotus-iroh: Please install a more recent version of Cockpit on Ubuntu 20.04 using "backports" with:

```sh
apt install -t focal-backports cockpit
```

(After running `apt update` to get the latest metadata, of course.)

This should upgrade you to Cockpit 252.

When running the above command to install Cockpit in this manner, on a fresh Ubuntu 20.04 test VM, it also happens to pull in `cockpit-networkmanager`, `cockpit-storaged`, and `cockpit-packagekit`, so this should hopefully get you on a much more recent version of Cockpit (as 215 is around a year and a half old and there have been several improvements since) and give you the pages within Cockpit that you'd like.

After running the command, please comment here again so we know it worked or you!

(Aside: We should specifically mention how to install Cockpit from backports on our website for Ubuntu. We do this for Debian, but Ubuntu doesn't say how to install from backports. Apologies for not having that in our docs already! I'll file an issue to get this fixed.)

**Comment by csjjpm** [comment]
Hi, I am having the same issue.  I have just replaced cockpit with the backports version.  I get a progress bar that shoots up and down and then finishes and I still only have the Storage application.
![tempsnip](https://user-images.githubusercontent.com/2385260/138095674-df27891e-4559-4c9a-a6d0-2023bf4a0004.png)

**Comment by garrett** [comment]
@csjjpm: Which version of Cockpit are you using, on which distribution and version of distribution? (I see you're talking about "backports", so I assume that means you're using some version of Debian or Ubuntu.)

We had an issue with progress bars going backwards in issue https://github.com/cockpit-project/cockpit/issues/13953, but that was supposedly fixed in https://github.com/cockpit-project/cockpit/pull/15584, which _should_ have landed in Cockpit 241, which was released on Mar 31, 2021. Which _should_ be in backports for Debian and Ubuntu already.

Knowing which distribution and version will help immensely. Thanks!

**Comment by martinpitt** [comment]
> I would very much like to check out Software Updates (PackageKit) and Network

You already have Software Updates, as the /apps page is shipped by the same Ubuntu package (cockpit-packagekit) as Software Updates.

Network does not have an "Applications" entry, you need to install this manually: `sudo apt install cockpit-networkmanager`.
